app/services/prisoner_slot_availability.rb

```diff
@@ -11,17 +12,24 @@ def initialize(prison, noms_id, date_of_birth,
     self.api_error     = false
   end
 
+  # rubocop:disable Metrics/MethodLength
   def slots
+    return all_slots unless in_trial?
     return all_slots if enforce_all_available_slots?
 
     results = all_slots.deep_dup.each { |slot, unavailability_reasons|
-      unless offender_availabilities_dates.include?(slot.to_date)
+      slot_date = slot.to_date
+      unless offender_availabilities_dates.include?(slot_date)
         unavailability_reasons << PRISONER_UNAVAILABLE
       end
-    }
 
+      unless bookable_prison_slots.include?(slot_date)
```

Comment:
For prison availability we want to check against the specific slot time instead of the slot date as the granularity is per slot.